### PR TITLE
Make RazorViewEngineOptions.FileProvider a list instead of a single item

### LINKED ISSUE #3806: Make RazorViewEngineOptions.FileProvider a list instead of a single item
See comment here: https://github.com/aspnet/Mvc/issues/3750#issuecomment-164474888

This has become a pretty common task for building anything modular. It would be nice if we just make `FileProvider` into a list so it was obvious that you can do this.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3806

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs [hunk 1]
```diff
@@ -97,9 +88,31 @@ public CSharpCompilationOptions CompilationOptions
...
                 _compilationOptions = value;
             }
         }
 
+        /// <summary>
+        /// Gets the composite result of <see cref="FileProviders"/>.
+        /// </summary>
+        /// <returns>The <see cref="IFileProvider"/> instance if exactly one instance is registered, otherwise
+        /// a <see cref="CompositeFileProvider"/> that contains all registered instances.</returns>
+        public IFileProvider GetCompositeFileProvider()
```
Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: Should we just move this to a facade service? I don't think this is horrid, just kinda ugly to put on options (which the user will see).

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: Ok. It seemed a bit easy to get away with sticking it on options. I can move it to a service if you feel strongly about it

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: I'm just asking the question. I think either one is a bit cumbersome. Let me ask @lodejard if he's around today.

Comment (davidfowl) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: This is a weird thing to expose

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: Fine. I'll make it a service.

Comment (davidfowl) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: Internal namespace?

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.Razor/RazorViewEngineOptions.cs]: That seems worse.

**Comment by pranavkm** [comment]
:up: :date:

**Comment by rynowak** [comment]
:shipit:

**Pull request closed without merging**